training/train.py

```diff
@@ -354,59 +320,123 @@ def get_deployed_model_version(yaml_data, deployment_prefix):
                 model_version = env.get('parameters', {}).get('MODEL_VERSION', '')
                 break
     else:
-        _logger.error('ERROR - Invalid deployment prefix: "{}", supported values: {}'.format(
-            deployment_prefix, list(DEPLOYMENT_PREFIX_ENVIRONMENT_NAME_MAP.keys())))
+        raise Exception(f'Invalid deployment prefix: "{deployment_prefix}", supported '
+                        f'values: {list(DEPLOYMENT_PREFIX_ENVIRONMENT_NAME_MAP.keys())}')
 
-    _logger.info('Model version: {} for deployment prefix: {}'.format(
-        model_version, deployment_prefix))
+    _logger.info('Model version: %s for deployment prefix: %s', model_version, deployment_prefix)
     return model_version
 
 
+def update_yaml_data(yaml_dict, deployment_prefix, model_version, hyper_params):
```

Comment:
Isn't better to take resolved deployment prefix as an argument instead of resolving inside the func?